compiler/rustc_hir_analysis/src/check/check.rs

```diff
@@ -235,16 +235,33 @@ pub(super) fn check_opaque_for_cycles<'tcx>(
     span: Span,
 ) -> Result<(), ErrorGuaranteed> {
     let args = GenericArgs::identity_for_item(tcx, def_id);
-    if tcx.try_expand_impl_trait_type(def_id.to_def_id(), args).is_err() {
-        let reported = opaque_type_cycle_error(tcx, def_id, span);
-        Err(reported)
-    } else if let Err(&LayoutError::Cycle(guar)) =
-        tcx.layout_of(tcx.param_env(def_id).and(Ty::new_opaque(tcx, def_id.to_def_id(), args)))
+
+    // First, try to look at any opaque expansion cycles, considering coroutine fields
+    // (even though these aren't necessarily true errors).
+    if tcx
```

Comment:
could we alternatively completely skip checking coroutines here? 🤔 it feels a bit off for me to mix coroutine and opaque type cycle checking as they are pretty much fundamentally different checks

e.g. i could imagine a separate global query check which eagerly computes the layout of coroutines